### DISCUSSION TITLE
Should http-client be supporting `application/problem+json` ?

### DISCUSSION THREAD

**Discussion by nyilmaz** [micronaut-projects/micronaut-core#6144] [Ideas]
Micronaut has already been supporting problem spec with https://github.com/micronaut-projects/micronaut-problem-json, indeed this is server side. What about http clients? Should clients understand `problem+json` and respond (or rethrow maybe) accordingly, instead of simply throwing `HttpClientResponseException`?

**Comment by jameskleeh** [comment]
We wouldn't change the behavior of the client depending on the content type

**Comment by nyilmaz** [comment]
@jameskleeh I see similar behavior [here](https://github.com/micronaut-projects/micronaut-core/blob/3.0.x/http-client-core/src/main/java/io/micronaut/http/client/exceptions/HttpClientErrorDecoder.java#L68-L75). Argument can be resolved by adding another chack if content-type is `application/problem+json`.
